File: plugins/buddy-pet/scripts/birth.py
#!/usr/bin/env python3
"""
Birth script for buddy-pet plugin.
Reads companion from ~/.claude.json, computes bones, registers buddy via API.
"""

import os
import sys

# Add scripts dir to path for common import
sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))
import common


def main():
    # 1. Check if already registered
    config = common.load_config()
    if config and config.get("buddy_token"):
        buddy_id = config.get("buddy_id", "unknown")
        print(f"Already registered. Buddy ID: {buddy_id}. Use /buddy-delete first.")
        return

    # 2. Read companion data from ~/.claude.json
    companion = common.read_companion_data()
    if not companion:
        print("No companion found in ~/.claude.json. Hatch a companion in Claude Code first.")
        return

    user_id = common.get_user_id()
    if user_id == "anon":
        print("Could not determine user ID from ~/.claude.json. Sign in to Claude Code first.")
        return

    # 3. Compute hashes
    user_hash = common.compute_user_hash(user_id)
    instance_id = common.compute_instance_id()

    # 4. Build birth request
    birth_data = {
        "user_hash": user_hash,
        "instance_id": instance_id,
        "companion": {
            "name": companion.get("name", ""),
            "personality": companion.get("personality", ""),
            "hatched_at": companion.get("hatched_at", 0),
            "species": companion.get("species", ""),
            "rarity": companion.get("rarity", ""),
            "eye": companion.get("eye", ""),
            "hat": companion.get("hat", ""),
            "shiny": companion.get("shiny", False),
            "base_stats": companion.get("base_stats", {}),
        },
        "plugin_version": common.PLUGIN_VERSION,
    }

    # 5. POST /guild/buddy/birth -> get buddy_token directly
    print("Registering buddy...")
    status, resp = common.http_post("/guild/buddy/birth", birth_data)

    if status == 0:
        print(f"Failed to connect to API: {resp.get('error', 'unknown error')}")
        return

    if status not in (200, 201):
        error = resp.get("error", resp.get("message", f"HTTP {status}"))
        print(f"Birth failed: {error}")
        return

    buddy_token = resp.get("buddy_token", "")
    buddy_id = resp.get("buddy_id", "")

    if not buddy_token:
        print(f"Unexpected response (no buddy_token): {resp}")
        return

    # 6. Save config
    api_url = common.get_api_url()
    new_config = {
        "buddy_token": buddy_token,
        "buddy_id": buddy_id,
        "api_url": api_url,
    }
    common.save_config(new_config)

    name = companion.get("name", "buddy")
    buddy_name = resp.get("buddy", {}).get("name", name)
    print(f"BUDDY born! Name: {buddy_name}, ID: {buddy_id}")


if __name__ == "__main__":
    main()

File: plugins/buddy-pet/scripts/birth.js
#!/usr/bin/env node
/**
 * Birth script for buddy-pet plugin.
 * Registers a new buddy on the social network.
 */
'use strict';

const common = require('./common');

async function main() {
  const config = common.loadConfig();
  if (config && config.buddy_token) {
    console.log(`Already registered. Buddy ID: ${config.buddy_id}`);
    process.exit(0);
  }

  const companion = common.readCompanionData();
  if (!companion) {
    console.log('No companion found in ~/.claude.json. Start a Claude Code session first.');
    process.exit(1);
  }

  const userId = common.getUserId();
  if (userId === 'anon') {
    console.log('Cannot determine user ID. Log in to Claude Code first.');
    process.exit(1);
  }

  const userHash = common.computeUserHash(userId);
  const instanceId = common.computeInstanceId();

  const birthData = {
    user_hash: userHash,
    instance_id: instanceId,
    companion: {
      name: companion.name,
      personality: companion.personality,
      hatched_at: companion.hatched_at,
      species: companion.species,
      rarity: companion.rarity,
      eye: companion.eye,
      hat: companion.hat,
      shiny: companion.shiny,
      base_stats: companion.base_stats,
    },
    plugin_version: common.PLUGIN_VERSION,
  };

  const [status, resp] = await common.httpPost('/guild/buddy/birth', birthData);

  if (status === 200 || status === 201) {
    const token = resp.buddy_token;
    const buddyId = resp.buddy_id;
    common.saveConfig({
      buddy_token: token,
      buddy_id: buddyId,
      api_url: common.getApiUrl(),
    });
    const name = resp.buddy && resp.buddy.name || companion.name;
    console.log(`\u2728 ${name} is born! ID: ${buddyId}`);
  } else {
    const err = resp.error;
    const msg = typeof err === 'object' ? (err.message || err.code || JSON.stringify(err)) : String(err || `HTTP ${status}`);
    console.log(`Error: ${msg}`);
    process.exit(1);
  }
}

main().catch((err) => { console.log('Error: ' + err.message); process.exit(1); });
